### PR TITLE
Configure and README improvements

### PR DESCRIPTION
**Pull request by nicolasvasilache**

### CODE REVIEW

**Review by gysit — APPROVED** [review]
LGTM!

File: configure.py
Comment (giuseros) [inline configure.py:33]: Minor issue: if the env var has a trailing `\' this will error

File: configure.py
Comment (giuseros) [inline configure.py:79]: This is problematic: I have different clones of llvm, and llvm-project is not the one I am using for this. Can we use an env var also for the path to LLVM? Or at least have the path to be specified as an option?

**Review by giuseros — COMMENTED** [review]
It works almost out-of-the-box for me. I have highlighted few issues I still had to build on AArch64.

File: configure.py
Comment (nicolasvasilache) [reply configure.py:33]: ok, but is this something to act on?
I don't plan on handling error cases here, just don't put a trailing char?

File: configure.py
Comment (nicolasvasilache) [reply configure.py:79]: I'll have to defer to @stellaraccident on this as the README and the script were inconsistent.
This has to do with how external projects build with LLVM and I think there were changes recently.
An easy way would be for you to put a symlink and call it a day.

File: configure.py
Comment (nicolasvasilache) [reply configure.py:121]: this was already here before and it works for me; does it break for you?

File: configure.py
Comment (nicolasvasilache) [reply configure.py:122]: not a problem on my machine either but ok

File: configure.py
Comment (nicolasvasilache) [reply configure.py:134]: works for me, updating.

File: configure.py
Comment (nicolasvasilache) [reply configure.py:122]: actually no, double quotes yiekd this error:

```
CMake Error at CMakeLists.txt:756 (message):
  The target `"X86"' is experimental and must be passed via
  LLVM_EXPERIMENTAL_TARGETS_TO_BUILD.
```

File: configure.py
Comment (stellaraccident) [reply configure.py:82]: It's your project to decide. I personally prefer having options to specify the paths like the script originally had. I don't have a strong reason -- just a lot of time making scripts like this more general for weird situations. If it were me, I would use the side by side checkout thing as a default but slow override with options.

File: configure.py
Comment (stellaraccident) [inline configure.py:34]: Do you really need this? It is redundant with the location of the script: os.path.dirname(__file__)

File: configure.py
Comment (stellaraccident) [inline configure.py:37]: I generally prefer better defaults (ie. Build for in source tree) that allows override but up to you.

File: configure.py
Comment (stellaraccident) [reply configure.py:124]: I think it gets escaped properly down the line by subprocess. Generally, you can count on each argument being an actual argument in the executable argv

File: configure.py
Comment (nicolasvasilache) [reply configure.py:82]: I don't have a strong opinion on this, I would just like consistency between LLVM, the README, what configure.py does and my ability to build locally. 

The current proposed changes are my iteration on it as I was not able to reconcile things otherwise.
I am happy to revert to a prior version if all things agree.

It is my understanding we have the constraint that iree_llvm_sandbox, llvm_project and iree must be silblings under a common parent root (or we had that constraint until recently). This seems to directly contradict being able to use iree/llvm-project, so this can't be true.

Anyway, I am looking for advice / what to revert while still being able to build and run the project.

File: configure.py
Comment (giuseros) [reply configure.py:124]: I was manually adding quotes before :). To me it breaks with: 
```
/bin/sh: clang: command not found
/bin/sh: clang-tools-extra: command not found
```
Might be because of a different shell?

Comment (stellaraccident) [reply configure.py:124]: Weird. If the quotes help and don't break the CI, fine with me (I've seen cases where adding quotes to this API does break things).

File: configure.py
Comment (giuseros) [reply configure.py:82]: The main problem is that in the same folder I have at least three llvm clones, and one of them is named llvm-project, but it is quite old, so I was getting weird errors (albeit the `configure.py` was still detecting the `llvm-project` folder) . I agree with @stellaraccident , having an additional option to override the default llvm-project path should be simple enough and unharmful. I don't want to block this PR though. I can add it later in a different PR

Comment (stellaraccident) [reply configure.py:82]: My guidance would be: the side by side constraint is a useful default, but explicit paths/options always gets used eventually for advanced cases.

The property that you really want is: check things out like this, run script without options, success. Having knobs to control it for advanced cases usually is required eventually, and there is nothing in the build setup that *requires* these projects live side by side.

File: configure.py
Comment (nicolasvasilache) [reply configure.py:34]: It is redundant but then the README is off.
I'd happily drop from the README `IREE_LLVM_SANDBOX_SOURCE_DIR`, the fewer chances for inconsistencies the better.

File: configure.py
Comment (nicolasvasilache) [reply configure.py:82]: `there is nothing in the build setup that requires these projects live side by side` ok I think this is the information I was hung up on, I should then be able to update more of the README and less of the configure script. I'll start a new PR.

File: configure.py
Comment (giuseros) [reply configure.py:36]: Well, it should be sufficient to replace it with:
```
os.getenv("IREE_LLVM_SANDBOX_SOURCE_DIR").rstrip('/') == os.path.abspath(os.path.dirname(__file__))
```

File: configure.py
Comment (giuseros) [reply configure.py]: Oh. What shell are you using ? On mine `-DLLVM_TARGETS_TO_BUILD="X86"` works fine

**Comment by nicolasvasilache** [comment]
@stellaraccident @giuseros I think I can retire this in favor of a simpler #19 sorry for the noise..

**Pull request closed without merging**
